rust-sdk/whirlpool/src/tests/rpc.rs

```diff
@@ -306,14 +351,108 @@ async fn send(
                 feature_set: Some(version.feature_set),
             })?
         }
+        "getProgramAccounts" => {
```

Comment:
I think this might be a slippery slope to do our own implementation of `getProgramAccounts`. This opens us up to several vectors where we might mess up if we forget something:
* Forget to add an account to the accounts hashmap (which is currently already happening for all the accounts initialized in tests/token.rs and tests/program.rs)
* Our implementation of memcmp might differ from the actual rpc-implementation (and there is currently no way of knowing if it does).

With this I am leaning to just doing what we did in in the ts tests: skip the tests that require gpa and add a comment saying that BanksClient doesn't support gpa and therefor we decided to leave that function untested

Reply:
Good point, I get where you're coming from. I initially thought it’d be fine since this is just for testing, but yeah, doing our own GPA implementation does feel a bit risky—especially if we accidentally miss things or introduce inconsistencies with how the actual RPC works.

I’m okay with the idea of doing what the TS tests do for now: skip the GPA-dependent tests and add a comment explaining the limitation with BanksClient. It’s not ideal, but it’s clear and avoids the hassle of maintaining a custom implementation.